pkg/liqo-controller-manager/resource-request-controller/local-resource-monitor.go

```diff
@@ -32,27 +31,22 @@ import (
 	"k8s.io/klog/v2"
 	resourcehelper "k8s.io/kubectl/pkg/util/resource"
 
-	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
 	"github.com/liqotech/liqo/pkg/consts"
-	"github.com/liqotech/liqo/pkg/liqo-controller-manager/resource-request-controller/interfaces"
 	"github.com/liqotech/liqo/pkg/utils"
 	liqoerrors "github.com/liqotech/liqo/pkg/utils/errors"
 	"github.com/liqotech/liqo/pkg/utils/pod"
 	"github.com/liqotech/liqo/pkg/virtualKubelet/forge"
 )
 
-// Broadcaster is an object which is used to get resources of the cluster.
-type Broadcaster struct {
-	allocatable               corev1.ResourceList
-	resourcePodMap            map[string]corev1.ResourceList
-	lastReadResources         map[string]corev1.ResourceList
-	nodeMutex                 sync.RWMutex
-	podMutex                  sync.RWMutex
-	nodeInformer              cache.SharedIndexInformer
-	podInformer               cache.SharedIndexInformer
-	updater                   interfaces.UpdaterInterface
-	resourceSharingPercentage uint64
-	updateThresholdPercentage uint64
+// LocalResourceMonitor is an object that keeps track of the cluster's resources.
+type LocalResourceMonitor struct {
+	allocatable    corev1.ResourceList
+	resourcePodMap map[string]corev1.ResourceList
+	nodeMutex      sync.RWMutex
+	podMutex       sync.RWMutex
+	nodeInformer   cache.SharedIndexInformer
+	podInformer    cache.SharedIndexInformer
+	notifier       ResourceUpdateNotifier
```

Comment:
Could it make sense to make this a list of notifiers?

Reply:
The need for multiple notifiers in parallel was not mentioned in the meeting. I can see why they could be useful, but it shouldn't be hard to implement them when needed.

Reply:
Yep, it should be trivial to add